main.go

```diff
@@ -62,18 +64,33 @@ func main() {
 	var rulesConfigmapName string
 	var rulesConfigmapNamespace string
 	var rulesFileName string
+	var rulesFilePath string
+
+	var operator controllers.OperatorMode
+
+	controllerCommand := flag.NewFlagSet("controller", flag.ExitOnError)
+	sidecarCommand := flag.NewFlagSet("sidecar", flag.ExitOnError)
 
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
-	flag.StringVar(&rulesConfigmapName, "rulesConfigmapName", "oathkeeper-rules", "Name of the Configmap that stores Oathkeeper rules.")
-	flag.StringVar(&rulesConfigmapNamespace, "rulesConfigmapNamespace", "oathkeeper-maester-system", "Namespace of the Configmap that stores Oathkeeper rules.")
 	flag.StringVar(&rulesFileName, "rulesFileName", "access-rules.json", "Name of the file with converted Oathkeeper rules")
 
+	controllerCommand.StringVar(&rulesConfigmapName, "rulesConfigmapName", "oathkeeper-rules", "Name of the Configmap that stores Oathkeeper rules.")
+	controllerCommand.StringVar(&rulesConfigmapNamespace, "rulesConfigmapNamespace", "oathkeeper-maester-system", "Namespace of the Configmap that stores Oathkeeper rules.")
+
+	sidecarCommand.StringVar(&rulesFilePath, "rulesFilePath", "/etc/config/access-rules.json", "Path to the file with converted Oathkeeper rules")
```

Comment:
Isn't this clashing with `rulesFileName`?

Reply:
No, because we used `NewFlagSet`, so the controller and sidecar modes can have different sets of flags, that won't mix :)

Reply:
Ah I see! Shouldn't we move `rulesFileName` to `controllerCommand`? It appears it's only being used in controller mode?

Reply:
Oh, yeah, missed that